lib/Relation.php

```diff
@@ -106,6 +107,89 @@ public function none(): Relation
         return $this;
     }
 
+    /**
+     * Enables the use of dynamic finders.
+     *
+     * Dynamic finders are just an easy way to do queries quickly without having to
+     * specify an options array with conditions in it.
+     *
+     * ```
+     * SomeModel::select('author_id')->find_by_first_name('Tito');
+     * SomeModel::select('author_id')->find_by_first_name_and_last_name('Tito','the Grief');
+     * SomeModel::select('author_id')->find_by_first_name_or_last_name('Tito','the Grief');
+     * ```
+     *
+     * You can also create the model if the find call returned no results:
+     *
+     * ```
+     * Person::find_or_create_by_name('Tito');
+     *
+     * # would be the equivalent of
+     * if (!Person::select('author_id')->find_by_name('Tito'))
+     *   Person::create(['Tito']);
+     * ```
+     *
+     * Some other examples of find_or_create_by:
+     *
+     * ```
+     * Person::select('id')->find_or_create_by_name_and_id('Tito',1);
+     * Person::select('id')->find_or_create_by_name_and_id(['name' => 'Tito', 'id' => 1]);
+     * ```
+     *
+     * @param $method Name of method
+     * @param $args   Method args
+     *
+     * @throws ActiveRecordException
+     *
+     * @see find
+     */
+    public function __call(string $method, mixed $args): mixed
```

Comment:
If you're going to move the docs here, then please remove them from `Model` so they're not repeated.

Reply:
Yeah, documentation needs work. We're inconsistent all over the place. I think once a documentation generator is chosen and put in place, then we can clean up things, remove duplication, etc. In this particular case for example, should Model refer to Relation's function header or vice versa. And why? What's the convention or reasoning?

Reply:
Well, the only problem I see is that all of the relation-forwarded calls on `Model` are static, whereas the methods on `Relation` aren't, so the samples we show are a bit weird. But, if we have that problem then so does Ruby on Rails. They seem to gloss over it, so I suppose we can, as well, but I suppose it wouldn't hurt to make sure that our samples show a sprinkling of both usages (static and non-static).

Reply:
So, I guess for consistencie's sake, you can go ahead and remove that block on `Model`, like we do for everything else that delegates to `Relation`.

Reply:
Done